controller/tenant_all.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"os"
+	"strings"
+
+	"github.com/fabric8-services/fabric8-tenant/app"
+	"github.com/fabric8-services/fabric8-tenant/jsonapi"
+	"github.com/fabric8-services/fabric8-tenant/keycloak"
+	"github.com/fabric8-services/fabric8-tenant/openshift"
+	"github.com/fabric8-services/fabric8-tenant/tenant"
+	"github.com/fabric8-services/fabric8-wit/errors"
+
+	"github.com/goadesign/goa"
+)
+
+// TenantAllController implements the tenantAll resource.
+type TenantAllController struct {
+	*goa.Controller
+	tenantService   tenant.Service
+	keycloakConfig  keycloak.Config
+	openshiftConfig openshift.Config
+	templateVars    map[string]string
+	usersURL        string
+}
+
+func CheckAccess(allowedusername, env string, s tenant.Service) error {
+	allowed := false
+	for _, username := range strings.Split(strings.TrimSpace(env), ",") {
+		if username == allowedusername {
+			allowed = true
+		}
+	}
+
+	if !allowed {
+		return errors.NewUnauthorizedError("you are not allowed to access to private apis request")
+	} else {
+		return nil
+	}
+}
+
+func NewTenantAllController(service *goa.Service, tenantService tenant.Service, keycloakConfig keycloak.Config, openshiftConfig openshift.Config, templateVars map[string]string) *TenantAllController {
+	return &TenantAllController{
+		Controller:      service.NewController("TenantAllController"),
+		tenantService:   tenantService,
+		keycloakConfig:  keycloakConfig,
+		openshiftConfig: openshiftConfig,
+		templateVars:    templateVars,
+	}
+}
+
+// Show runs the show action.
+func (c *TenantAllController) Show(ctx *app.ShowTenantAllContext) error {
+	if len(ctx.Request.Header["Authorization"]) == 0 {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("Expecting Authorization Header"))
+	}
+	aheader := strings.Split(ctx.Request.Header["Authorization"][0], " ")
+	if len(aheader) != 2 || strings.ToUpper(aheader[0]) != "BEARER" {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("Invalid Authorization Header"))
+	}
+	osconfig := c.openshiftConfig
+	osconfig.Token = aheader[1]
+	osusername, err := openshift.WhoAmI(osconfig)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("Cannot check OpenShift identity"))
+	}
+
+	err = CheckAccess(osusername, os.Getenv("F8_TENANT_ACCESS_LIST_TENANT_NAMESPACES"), c.tenantService)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(
+			"You don't have access to namespaces listing"))
+	}
+
+	namespaces, err := c.tenantService.GetAllNamespaces()
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+
+	response := app.TenantAll{
+		Namespaces: []*app.NamespaceAttributes{},
+	}
+
+	for _, ns := range namespaces {
```

Comment:
Why a List of Namespaces and not a List of Tenants?

Reply:
because I can only see the IDs and emails in the `tenants` table and not the names, or maybe I am missing something? I can do some strings.Split over the namespace.name but I thought that while at it we may want to have all the namespaces created by Fabric8 instead of just the tenant names. The call name may be a bit misleading and can rename that (although not sure to waht)

Reply:
Would just renewing this to 👍 
/api/namespaces/all

woudl work for you?